test/unit/controller/ResultControllerTest.php

```diff
@@ -0,0 +1,198 @@
+    {
+        $requestXml = '<?xml version="1.0" encoding="UTF-8"?>
+            <imsx_POXEnvelopeRequest xmlns="http://www.imsglobal.org/services/ltiv1p1/xsd/imsoms_v1p0">
+                <imsx_POXHeader>
+                    <imsx_POXRequestHeaderInfo>
+                        <imsx_version>V1.0</imsx_version>
+                        <imsx_messageIdentifier>999999123</imsx_messageIdentifier>
+                    </imsx_POXRequestHeaderInfo>
+                </imsx_POXHeader>
+                <imsx_POXBody>
+                    <replaceResultRequestFake>
+                        <resultRecord>
+                            <sourcedGUID>
+                                <sourcedId>3124567</sourcedId>
+                            </sourcedGUID>
+                            <result>
+                                <resultScore>
+                                    <language>en</language>
+                                    <textString>{{score}}</textString>
+                                </resultScore>
+                            </result>
+                        </resultRecord>
+                    </replaceResultRequestFake>
+                </imsx_POXBody>
+            </imsx_POXEnvelopeRequest>
+        ';
+
+        $subject = $this->getResultController();
+        $result = $subject->manageResult($requestXml);
+
+        $this->assertInstanceOf(Response::class, $result);
+        $this->assertEquals(501, $result->getStatusCode());
+    }
+
+    public function testManageResultWithIncorrectPayload()
+    {
+        $requestXml = '';
+
+        $subject = $this->getResultController();
+        $result = $subject->manageResult($requestXml);
+
+        $this->assertInstanceOf(Response::class, $result);
+        $this->assertEquals(501, $result->getStatusCode());
+    }
+
+    public function queryScoresToTest()
+    {
+        return [
+            ['{{score}}', '-1', 400],
+            ['{{score}}', 'string', 400],
+            ['{{score}}', '2', 400],
+            ['{{score}}', '0.92', 201],
+            [['{{score}}', '3124567'], ['0.92', '3124568'], 404],
+        ];
+    }
+
+    /**
+     * @dataProvider queryScoresToTest
+     *
+     * @param $search
+     * @param $score
+     * @param $expectedStatus
+     */
+    public function testManageResultWithScores($search, $score, $expectedStatus)
+    {
+        $requestXml = str_replace($search, $score, self::PAYLOAD_TEMPLATE);
+
+        $subject = $this->getResultController();
+        $result = $subject->manageResult($requestXml);
+        $this->assertInstanceOf(Response::class, $result);
+        $this->assertEquals($expectedStatus, $result->getStatusCode());
+    }
+
+    public function testDeliveryExecutionRetrieving()
+    {
+        $requestXml = str_replace('{{score}}', '0.92', self::PAYLOAD_TEMPLATE);
+
+        $subject = $this->getResultController();
+        $result = $subject->manageResult($requestXml);
+    }
+
+    private function getResultController()
```

Comment:
Make of this method the setUp method. It's called in all your tests to build the $subject.